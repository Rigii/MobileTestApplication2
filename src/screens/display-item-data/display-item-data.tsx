import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {connect} from 'react-redux';
import {ICONS} from '../../constants/icons';
import {COLORS} from '../../constants/theme';
import {STRINGS} from '../../constants/strings';
import {ROUTES} from '../../constants/routes';
import {deleteItem} from '../todo-list/todo.actions';
import SvgUri from 'react-native-svg-uri';

export const DisplayItemDataComp = (props: any) => {
  
  const navigationState = props.navigation.getParam('itemParams');
  const strings = STRINGS.DISPLAY_ITEM;

  //console.log(navigationState)

  const deleteItem = () => {
    props.navigation.navigate(ROUTES.TodoList);
    props.deleteItem(navigationState.id);
  };

  return (
    <View style={styles.add_item_view}>
      <View style={{alignItems: 'center'}}>
        <Text>{navigationState.title}</Text>
      </View>
      <View style={styles.stucture_comp}>
        <Text>{navigationState.description}</Text>
      </View>
      <View style={[styles.but_container, styles.stucture_comp]}>
        {[ICONS.location, ICONS.photo, ICONS.video].map((icon, index) => (
          <TouchableOpacity key={index} style={styles.button_show}>
          <SvgUri width="30" height="30" source={icon} />
        </TouchableOpacity>
        ))}
      </View>
      <TouchableOpacity
      testID="delItem"
        style={[styles.but, styles.stucture_comp]}
        onPress={deleteItem}>
        <Text>{strings.del_but}</Text>
      </TouchableOpacity>
    </View>
  );
};

const mapDispatchToProps = {
  deleteItem,
};

export const DisplayItemData = connect(
  null,
  mapDispatchToProps,
)(DisplayItemDataComp);

const styles = StyleSheet.create({
  add_item_view: {
    flex: 1,
    alignItems: 'center',
    // justifyContent: 'center',
    alignContent: 'center',
    margin: 15,
  },
  stucture_comp: {
    alignItems: 'center',
    marginTop: 20,
  },
  input: {
    width: 350,
    height: 40,
    borderColor: COLORS.input_border,
    borderWidth: 1,
    borderRadius: 5,
  },
  input_descript: {
    width: 350,
    height: 350,
    borderColor: COLORS.input_border,
    borderWidth: 1,
    borderRadius: 5,
  },
  text: {
    alignSelf: 'center',
    fontSize: 14,
  },
  button_show: {
    backgroundColor: COLORS.main,
    borderRadius: 50,
    padding: 15,
    opacity: 0.5,
    borderColor: COLORS.border_but_color,
  },
  but_container: {
    position: 'absolute',
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-around',
    bottom: 10,
  },
  but: {
    flex: 0,
    backgroundColor: COLORS.main,
    borderRadius: 5,
    padding: 15,
    paddingHorizontal: 20,
    alignSelf: 'center',
  },
});
