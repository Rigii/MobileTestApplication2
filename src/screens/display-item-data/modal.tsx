import React from 'react';
import { View, Modal } from 'react-native';

interface IItemModal {
    isVisible: boolean | undefined,
    content: React.ReactChild | null
}

export const ItemModal = (props: IItemModal) => {
    return(
        <View style={{marginTop: 22}}>
        <Modal
          animationType="fade"
          transparent={true}
          visible={props.isVisible}
          >
              {props.content}
        </Modal>
      </View>
    )

}