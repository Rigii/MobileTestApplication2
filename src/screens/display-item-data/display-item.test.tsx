import * as React from 'react';
import {DisplayItemDataComp} from './display-item-data';
import {render, fireEvent} from '@testing-library/react-native';

describe('Display item component', () => {
  const itemParams = {
    description:
      'Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem',
    id: 125,
    location: 'dfgdfg',
    photoUrl: 'dfgdg',
    title: 'Go to daddy',
    video: 'dfgfg',
  };

  const props = {
    deleteItem: jest.fn(),
    navigation: {
      navigate: jest.fn(),
      getParam: jest.fn(() => itemParams), // Using a mock function (mock params)
    } as any,
  };

  const {getByTestId} = render(<DisplayItemDataComp {...props} />);

  it('Should match to DisplayItemDataComp snapshot', () => {
    const {asJSON} = render(<DisplayItemDataComp {...props} />);
    expect(asJSON()).toMatchSnapshot();
  });

  it('Should delete function if del button are pushed', () => {
    const delBut = getByTestId('delItem');
    fireEvent.press(delBut)
    expect(props.deleteItem).toHaveBeenCalled()
  });
  
});
