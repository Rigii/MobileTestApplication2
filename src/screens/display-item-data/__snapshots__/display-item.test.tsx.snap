// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display item component Should match to DisplayItemDataComp snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flex": 1,
        "margin": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text>
        Go to daddy
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 20,
        }
      }
    >
      <Text>
        Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "bottom": 10,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        style={
          Object {
            "backgroundColor": "orange",
            "borderColor": "gray",
            "borderRadius": 50,
            "opacity": 0.5,
            "padding": 15,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        style={
          Object {
            "backgroundColor": "orange",
            "borderColor": "gray",
            "borderRadius": 50,
            "opacity": 0.5,
            "padding": 15,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        style={
          Object {
            "backgroundColor": "orange",
            "borderColor": "gray",
            "borderRadius": 50,
            "opacity": 0.5,
            "padding": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "orange",
          "borderRadius": 5,
          "flex": 0,
          "marginTop": 20,
          "opacity": 1,
          "padding": 15,
          "paddingHorizontal": 20,
        }
      }
      testID="delItem"
    >
      <Text>
        DELETE ITEM
      </Text>
    </View>
  </View>
</View>
`;
